modules/apps/depot/depot-web/src/main/resources/META-INF/resources/js/addConnectedSiteButtonPropsTransformer.js

```diff
@@ -0,0 +1,42 @@
+export default function propsTransformer({
+	currentURL,
+	eventName,
+	itemSelectorURL,
+	portletNamespace,
```

Comment:
Now, this might be a bit of a double footgun.
1. If we were to destructure here `portletId` or `componentId` and forget to return them, [`render`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.es.js#L38) might be missing them and fail in different ways.
2. I think right now `portletNamespace` isn't really a `namespace` (see necessary underscores below), so we might need to change this 😭

Reply:
> 1. If we were to destructure here `portletId` or `componentId` and forget to return them, [`render`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.es.js#L38) might be missing them and fail in different ways.

True, but if you destructure those items and then neglect to pass them, it is a pretty explicit action.  ie. it is a footgun, but with a safety on it, and you have to submit a background check before they'll even let you hold it.

Reply:
Fair enough